lite2/client.go

```diff
@@ -146,6 +146,9 @@ type Client struct {
 	// See Witnesses option
 	witnesses []provider.Provider
 
+	// Mutex for locking during changes of the lite clients providers
+	providerMutex sync.Mutex
```

Comment:
the convention we're using in Tendermint - fields protected by the mutex should be declared after it. E.g.

```go
mtx      sync.Mutex
A typeA
B typeB

V typeC // fields, which are not protected by mutex, get separated by a new line
```